### PR TITLE
CDATASOURC-3: Fix JNDI issues in Datasources

### PR DESCRIPTION
**Pull request by ChanakaCooray**

### CODE REVIEW

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceJndiManager.java [hunk 1]
```diff
@@ -33,10 +33,9 @@
  */
 public class DataSourceJndiManager {
 
-    private static Logger logger = LoggerFactory.getLogger(DataSourceJndiManager.class);
-
     private static final String JAVA_COMP_CONTEXT_STRING = "java:comp";
     private static final String ENV_CONTEXT_STRING = "env";
+    private static Logger logger = LoggerFactory.getLogger(DataSourceJndiManager.class);
```
Comment (DilanUA) [inline components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceJndiManager.java:38]: You can set this logger var "final" too.

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/internal/DataSourceListenerComponent.java [hunk 2]
```diff
@@ -53,12 +54,25 @@
...
+            name = "org.wso2.carbon.datasource.jndi",
+            service = JNDIContextManager.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unRegisterJNDIContextManager"
+    )
+    protected void registerJNDIContextManager(JNDIContextManager jndiContextManager) {
+    }
+
+
```
Comment (DilanUA) [inline components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/internal/DataSourceListenerComponent.java:74]: remove unnecessary new lines

File: tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceManagementServiceTest.java [hunk 3]
```diff
@@ -54,7 +54,8 @@
                 mavenBundle().artifactId("HikariCP").groupId("com.zaxxer").version("2.4.1"),
                 mavenBundle().artifactId("h2").groupId("com.h2database").version("1.4.191"),
                 mavenBundle().artifactId("org.wso2.carbon.datasource.core").groupId("org.wso2.carbon.datasources")
-                        .versionAsInProject()
+                        .versionAsInProject(),
```
Comment (DilanUA) [inline tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceManagementServiceTest.java:57]: better to move . sign to the upper part of the statement to imply continuation.

File: tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceServiceTest.java [hunk 4]
```diff
@@ -47,7 +47,8 @@
                 mavenBundle().artifactId("HikariCP").groupId("com.zaxxer").version("2.4.1"),
                 mavenBundle().artifactId("h2").groupId("com.h2database").version("1.4.191"),
                 mavenBundle().artifactId("org.wso2.carbon.datasource.core").groupId("org.wso2.carbon.datasources")
-                        .versionAsInProject()
+                        .versionAsInProject(),
```
Comment (DilanUA) [inline tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceServiceTest.java:50]: better to move . sign to the upper part of the statement to imply continuation.

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceJndiManager.java (hunk 1 above)
Comment (ChanakaCooray) [reply components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceJndiManager.java:38]: Fixed in https://github.com/wso2/carbon-datasources/pull/28

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/internal/DataSourceListenerComponent.java (hunk 2 above)
Comment (ChanakaCooray) [reply components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/internal/DataSourceListenerComponent.java:74]: Fixed in https://github.com/wso2/carbon-datasources/pull/28

File: tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceManagementServiceTest.java (hunk 3 above)
Comment (ChanakaCooray) [reply tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceManagementServiceTest.java:57]: Line break should have to be before the dot. That is the java 8 standard we have followed. Example[1].

[1] https://github.com/wso2/carbon-kernel/blob/master/core/src/main/java/org/wso2/carbon/kernel/internal/startupresolver/StartupOrderResolver.java

File: tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceServiceTest.java (hunk 4 above)
Comment (ChanakaCooray) [reply tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceServiceTest.java:50]: Line break should have to be before the dot. That is the java 8 standard we have followed. Example[1].

[1] https://github.com/wso2/carbon-kernel/blob/master/core/src/main/java/org/wso2/carbon/kernel/internal/startupresolver/StartupOrderResolver.java

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceJndiManager.java (hunk 1 above)
Comment (DilanUA) [reply components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceJndiManager.java:38]: Approved...

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/internal/DataSourceListenerComponent.java (hunk 2 above)
Comment (DilanUA) [reply components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/internal/DataSourceListenerComponent.java:74]: Approved...

File: tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceManagementServiceTest.java (hunk 3 above)
Comment (DilanUA) [reply tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceManagementServiceTest.java:57]: Approved...

File: tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceServiceTest.java (hunk 4 above)
Comment (DilanUA) [reply tests/osgi-tests/src/test/java/org/wso2/carbon/datasource/osgi/DataSourceServiceTest.java:50]: Approved...

**Pull request merged**